.github/workflows/deploy-gh-pages.yml

```diff
@@ -15,13 +16,13 @@ jobs:
       - name: 🐍 Set up Python 3
         uses: actions/setup-python@v2
         with:
-          python-version: 3
+          python-version: '3.x'
           cache: "pip"
 
       - name: 👷‍ Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install -r requirements.txt
+          python -m pip install --upgrade -r requirements.txt
```

Comment:
The issue here is we're not installing in to a fresh venv, but rather the existing Python, so some deps could already exist either from being system-installed, and/or injected by default into the env. Including `--upgrade` ensures at least first-level deps get upgraded, if possible.

As a sidenote, for repos of Python packages, rather than docs, websites, etc like this one, I keep everything clean by using venvs/conda envs. To note, even when creating a fresh venv with `python -m venv`, cached copies of some existing deps, including `pip`, `setuptools` and `pkg_resources` get injected into it, instead of downloading fresh copies of `conda create`. So its a good idea to add `setuptools` and `wheel` to the first line to ensure they are available and up to date for non-PEP 517 installs, though I didn't add them here to avoid another big debate.